api/src/main/java/org/datacleaner/configuration/RemoteServerData.java

```diff
@@ -0,0 +1,37 @@
+/**
+ * DataCleaner (community edition)
+ * Copyright (C) 2014 Neopost - Customer Information Management
+ *
+ * This copyrighted material is made available to anyone wishing to use, modify,
+ * copy, or redistribute it subject to the terms and conditions of the GNU
+ * Lesser General Public License, as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this distribution; if not, write to:
+ * Free Software Foundation, Inc.
+ * 51 Franklin Street, Fifth Floor
+ * Boston, MA  02110-1301  USA
+ */
+package org.datacleaner.configuration;
+
+/**
+ * Box for data about remote server
+ * @since 15. 10. 2015
+ */
+public interface RemoteServerData {
+    public RemoteServerData setServerPriority(Integer serverPriority);
+    public Integer getServerPriority();
+    public RemoteServerData setServerName(String serverName);
+    public String getServerName();
+    public RemoteServerData setHost(String host);
+    public String getHost();
+    public RemoteServerData setUsername(String username);
+    public String getUsername();
+    public RemoteServerData setPassword(String password);
```

Comment:
There is a list of these instances in OptionsDialog. The first instance can be updated via GUI dialog dynamically which (according to me) requires at least setUsername() and setPassword() to be available.